consensus/src/block_storage/block_store.rs

```diff
@@ -381,7 +411,10 @@ impl BlockStore {
         match self.get_block(qc.certified_block().id()) {
             Some(executed_block) => {
                 ensure!(
-                    executed_block.block_info() == *qc.certified_block(),
+                    // decoupled execution allows dummy block infos
+                    executed_block
```

Comment:
hmm, why we need to change this? qc should always match the executed result regardless of dummy or not

Reply:
if the node changes decoupled config and restart, the consensus db might contain blocks with correct executed result?

Reply:
no, there should be a clear epoch boundary for that